impl/src/planning.rs

```diff
@@ -593,6 +618,9 @@ impl<'planner> CrateSubplanner<'planner> {
     let mut dev_dep_names = Vec::new();
     let mut normal_dep_names = Vec::new();
 
+    // Store renamed dependencies in a HashMap
+    let mut renamed_dep_names = HashMap::new();
```

Comment:
The variable is defined here because the usage is at the bottom of the `for` loop.

Reply:
(My suggestion here is to move this declaration to L656)

Reply:
[L656 in the commit this review is based on](https://github.com/google/cargo-raze/blob/5708a78840fd58389278de6c2695a9002a2bb9b8/impl/src/planning.rs#L656) is inside of the `for` loop. The logic looks weird, but that `map` (which has now been changed to `if let` syntax) was only to avoid unwrapping `dep.rename`. The HashMap needs to be declared above the `for` loop so that any renamed dependencies can be added to it. [The latest commit](https://github.com/google/cargo-raze/blob/993622e766ccad5f6693ef1e2ad2e414b80a7a11/impl/src/planning.rs#L612-L650) makes this logic much more readable.

Reply:
My bad, you're definitely right. Another useless comment from me, hah.

Marking resolved.

Reply:
No worries!